Source/Core/Core/PowerPC/JitArm64/Jit.cpp

```diff
@@ -239,26 +251,65 @@ void JitArm64::DumpCode(const u8* start, const u8* end)
 	WARN_LOG(DYNA_REC, "Code dump from %p to %p:\n%s", start, end, output.c_str());
 }
 
+void JitArm64::EmitResetCycleCounters()
+{
+	const u32 PMCR_EL0_E  = 1;
+	const u32 PMCR_EL0_P  = 2;
+	const u32 PMCR_EL0_C  = 4;
+	const u32 PMCR_EL0_LC = 0x40;
+	_MSR(FIELD_PMCR_EL0, X0);
+	MOVI2R(X1, PMCR_EL0_E |
+	           PMCR_EL0_P |
+	           PMCR_EL0_C |
+	           PMCR_EL0_LC);
+	ORR(X0, X0, X1);
+	MRS(X0, FIELD_PMCR_EL0);
+}
+
+void JitArm64::EmitGetCycles(Arm64Gen::ARM64Reg reg)
+{
+	_MSR(FIELD_PMCCNTR_EL0, reg);
+}
+
 void JitArm64::BeginTimeProfile(JitBlock* b)
 {
 	b->ticCounter = 0;
 	b->ticStart = 0;
 	b->ticStop = 0;
 
-	MOVI2R(X1, (u64)QueryPerformanceCounter);
-	MOVI2R(X0, (u64)&b->ticStart);
-	BLR(X1);
+	if (m_supports_cycle_counter)
+	{
+		EmitResetCycleCounters();
```

Comment:
The cycle counter can and will wrap around causing overflow issues later when we subtract the start and end times. So it needs to be reset constantly.